### PR TITLE
[docs] Fix minor verbiage and formatting issues in SSO guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Follow up for #25457 as I'm not sure why GitHub UI decided to not add my suggestion when reviewing the PR.

# How

- Use words instead of symbols
- Remove unnecessary callout. The example doesn't have to be a callout, it can be a part of the step itself. Also, in short span of sections, there are so many callouts, I just removed the unnecessary one and made the text part of the main flow.

CleanShot 2023-11-19 at 12 55 53@2x

- Apply Prettier formatting.

# Test Plan

Run docs locally and visit: http://localhost:3002/accounts/sso/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/accounts/sso.mdx [hunk 1]
```diff
@@ -56,15 +54,17 @@ SSO users are like regular users. However, there are a few known exceptions:
...
+3. By default, your new SSO user will have the View Only role. If you need a different role, ask an Admin or Owner to update your role in [member settings](https://expo.dev/accounts/[account]/settings/members).
+4. Run `eas login --sso` to switch to your new account on the CLI.
+5. At this time, the Admin or Owner can remove your old user from the organization. In [member settings](https://expo.dev/accounts/[account]/settings/members), the list of organization members indicates whether a user is an SSO or non-SSO user. The Admin or Owner can click the dropdown next to the old user and click **Remove member**.
+6. If you no longer need your old user account, log out of your new SSO account, then log in to your old account and go to [user settings](https://expo.dev/settings). Scroll down and click **Delete Account**. **Note that this will delete any projects under your old user account.** It will not affect any projects owned by the organization.
```
Comment (Simek) [inline docs/pages/accounts/sso.mdx:67]: Thanks for correcting this! We also could use `Step` component here in the future.

**Review by Simek — APPROVED** [review]
👌

File: docs/pages/accounts/sso.mdx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/accounts/sso.mdx:67]: Great catch! Let me update the PR.

Comment (amandeepmittal) [reply docs/pages/accounts/sso.mdx]: Formatted the whole guide to use `Step` component and also find another issue where an internal link was missing.

**Pull request merged**